[EJS] Add note about QuickJS executable name (#76)
yt-dlp/yt-dlp-wiki: EJS.md

## Step 1: Install a supported JavaScript Runtime

For yt-dlp to recognize your JavaScript Runtime, you need to have it inside a folder in your PATH, or, if on Windows, in the same folder as ``yt-dlp.exe``. If you don't, you will have to add the path to the directory containing it.

> 

Reviewer: I would also move this below the table since it references `--js-runtimes`, which is better introduced by the table
``````suggestion
The JavaScript runtime executable should be located within your `PATH` (or located in the same folder as your `yt-dlp.exe` if you are on Windows). Otherwise, you'll need to specify the path to the JS runtime executable (or its containing folder) in your `--js-runtimes` argument, e.g.:

```
--js-runtimes "deno:/path/to/deno"
```

``````

- QuickJS versions prior to `2025-4-26` are missing optimizations which can lead to execution times of several minutes.
- QuickJS-NG versions prior to `0.12.0 `are missing optimizations which can lead to execution times of several minutes.
- Both QuickJS and QuickJS-NG do not fully allow executing files from stdin, so yt-dlp will create temporary files for each EJS script execution. This can theoretically lead to time-of-check to time-of-use (TOCTOU) vulnerabilities.

> - The name for the executable must be `qjs` (`qjs.exe` in Windows) in order to be read by yt-dlp without specifying the path to it.

Reviewer: ```suggestion
- The basename of the executable must be `qjs` (or `qjs.exe` on Windows) in order for yt-dlp to find it within your `PATH`. Otherwise, you'll need to pass the path to the executable as part of the `--js-runtimes` argument.
```

Reviewer: ```suggestion
- The basename of the executable must be `qjs` (or `qjs.exe` on Windows) in order for yt-dlp to find it when no path is specified in the `--js-runtimes` argument.
```

Reviewer: ```suggestion
- The filename of the executable must be `qjs` (or `qjs.exe` on Windows) in order for yt-dlp to find it when no path is specified in the `--js-runtimes` argument.
```

Reviewer: ```suggestion
- The filename of the executable must be `qjs` (or `qjs.exe` on Windows), or else the path to the executable file must be specified in the `--js-runtimes` argument.
```

- QuickJS versions prior to `2025-4-26` are missing optimizations which can lead to execution times of several minutes.
- QuickJS-NG versions prior to `0.12.0 `are missing optimizations which can lead to execution times of several minutes.
- Both QuickJS and QuickJS-NG do not fully allow executing files from stdin, so yt-dlp will create temporary files for each EJS script execution. This can theoretically lead to time-of-check to time-of-use (TOCTOU) vulnerabilities.
- The filename of the executable must be `qjs` (or `qjs.exe` on Windows), or else the path to the executable file must be specified in the `--js-runtimes` argument.

> 

Reviewer: ```suggestion
- The filename of the executable must be `qjs` (or `qjs.exe` on Windows), or else the path to *the executable file* must be specified in the `--js-runtimes` argument.

```
